addons/l10n_cl/models/res_partner.py

```diff
@@ -21,3 +22,32 @@ class ResPartner(models.Model):
              '2 - Fees Receipt Issuer (Applies to suppliers who issue fees receipt)\n'
              '3 - End consumer (only receipts)\n'
              '4 - Foreigner')
+
+    def _format_vat_cl(self, values):
+        identification_types = [self.env.ref('l10n_latam_base.it_vat').id, self.env.ref('l10n_cl.it_RUT').id,
+                                self.env.ref('l10n_cl.it_RUN').id]
+        country_id = self.env.ref('base.cl').id
+        if values.get('country_id') and values.get('country_id') == country_id and \
+                values.get('l10n_latam_identification_type_id') and values.get('l10n_latam_identification_type_id') in \
+                identification_types and values.get('vat'):
+            return stdnum.util.get_cc_module('cl', 'vat').format(values['vat']).replace('.', '').replace(
+                'CL', '').upper()
+        else:
+            return values['vat']
+
+    @api.model
+    def create(self, values):
+        if values.get('vat') or values.get('l10n_latam_identification_type_id') or values.get('country_id'):
+            values['vat'] = self._format_vat_cl(values)
+        return super().create(values)
+
+    def write(self, values):
+        for record in self:
+            vat_values = {
+                'vat': values.get('vat', record.vat),
```

Comment:
@Danisan This is wrong.  Consider the case when you put the vat value (or any field) back to no value.  It should be 'latam_identification_type_id' in values and ... or

Reply:
cannot catch the case... I already tried in the interface to put some values in no values, and cannot see an error there.. besides latam_identification_type_id is mandatory.

Could we test in runbot?

Reply:
@jco-odoo I was checking this:
We need to pass the three values (vat, country and document type) to the format function. That is why they are passed using the vat_values dictionary.
The approach to what you say is something like:

```
vat_values = {}
if 'vat' in value:
    vat_value['vat'] = value['vat']
if 'country_id' in value:
    vat_value['country_id'] = value['country_id']
....
```

Finally, with this code it would be the same to pass "value" dictionary instead of passing "vat_value" dictionary, and this will be not useful, because we decide to apply the formatting based on the country and Latam id type.

So if we need the values in the function, even if it has not been modified I need to pass them to formatting function.

Take a look also, that if the values are not there (empty for example) the _format_vat_cl returns the vat and there should be no error, as we could prove today with the examples.

Reply:
`{'vat': 'vat' in values and values['vat} or rec.vat, 
  'latam_identification_type_id': 'latam_identiifcation_type_id" in values and values['latam'] or rec.latam_identification_type_id, ....`

Reply:
I am wrong apparently. values.get( ,) always gives the value even if False.

Reply:
So this is correct